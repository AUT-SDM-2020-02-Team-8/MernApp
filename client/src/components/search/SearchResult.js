import React, { Component } from "react"
import SearchResultRow from "./SearchResultRow"
import { TableContainer, Table, TableHead, TableBody, TableRow, TableCell } from "@material-ui/core"

class SearchResult extends Component {
  render() {
    return (
      <TableContainer>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Title</TableCell>
              <TableCell>Author</TableCell>
              <TableCell>Year</TableCell>
              <TableCell>Type</TableCell>
              <TableCell>Journal</TableCell>
              <TableCell>Publisher</TableCell>
              <TableCell>SE Practice</TableCell>
              <TableCell>Claims</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {this.props.evidences.map(evidence => {
              return ( <SearchResultRow key={evidence._id} evidence={evidence} /> )
            })}
          </TableBody>
        </Table>
      </TableContainer>
    )
  }
}

export default SearchResult