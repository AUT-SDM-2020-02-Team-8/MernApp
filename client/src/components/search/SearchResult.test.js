import React from 'react';
import { shallow } from 'enzyme';
import SearchResult from './SearchResult';

test('renders homepage', () => {
  const evidences = [
    {
      _id: '1',
      title: 'title 1',
      author: 'author 1',
      year: '2020',
      recordType: 'reference paper',
      journal: 'IEEE',
      publisher: 'IEEE',
      sePractice: 'TDD',
      claims: ["code quality", "team confidence", "product quality"]
    }
  ];
  const wrapper = shallow(<SearchResult evidences={evidences} />);
  expect(wrapper).toMatchSnapshot();
});
