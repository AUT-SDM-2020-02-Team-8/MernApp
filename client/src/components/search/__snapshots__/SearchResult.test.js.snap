// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage 1`] = `
<WithStyles(ForwardRef(TableContainer))>
  <ShowColumnsFilter
    showColumns={[Function]}
  />
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableCell))
          key="levelOfSupport"
          onClick={[Function]}
        >
          Level of Support
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="title"
          onClick={[Function]}
        >
          Title
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="author"
          onClick={[Function]}
        >
          Author
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="year"
          onClick={[Function]}
        >
          Year
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="recordType"
          onClick={[Function]}
        >
          Type
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="journal"
          onClick={[Function]}
        >
          Journal
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="publisher"
          onClick={[Function]}
        >
          Publisher
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="sePractice"
          onClick={[Function]}
        >
          SE Practice
            
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          key="claims"
          onClick={[Function]}
        >
          Claims
            
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody))>
      <SearchResultRow
        chosenColumns={
          Array [
            "levelOfSupport",
            "title",
            "author",
            "year",
            "recordType",
            "journal",
            "publisher",
            "sePractice",
            "claims",
          ]
        }
        evidence={
          Object {
            "_id": "1",
            "author": "author 1",
            "claims": Array [
              "code quality",
              "team confidence",
              "product quality",
            ],
            "journal": "IEEE",
            "publisher": "IEEE",
            "recordType": "reference paper",
            "sePractice": "TDD",
            "title": "title 1",
            "year": "2020",
          }
        }
        key="1"
      />
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
</WithStyles(ForwardRef(TableContainer))>
`;
